src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSFunctionBinding.cs

```diff
@@ -220,15 +217,14 @@ internal static unsafe void InvokeJSFunction(JSObject jsFunction, Span<JSMarshal
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal static unsafe void InvokeJSImportImpl(JSFunctionBinding signature, Span<JSMarshalerArgument> arguments)
         {
+#if FEATURE_WASM_THREADS
+            JSProxyContext.AssertCurrentContext();
+#endif
+
             if (signature.IsAsync)
             {
                 // pre-allocate the result handle and Task
-#if FEATURE_WASM_THREADS
-                JSSynchronizationContext.AssertWebWorkerContext();
-                var holder = new JSHostImplementation.PromiseHolder(JSSynchronizationContext.CurrentJSSynchronizationContext!);
-#else
-                var holder = new JSHostImplementation.PromiseHolder();
-#endif
+                var holder = new JSHostImplementation.PromiseHolder(JSProxyContext.DefaultInstance);
```

Comment:
DefaultInstance is a weird name for this. I would prefer a name that conveys more meaning like 'the instance for the current operation' or 'the instance for the current thread'. In this case if we're invoking a JSImport on the current thread, shouldn't it be the context for the current thread? Looking at the definition of DefaultInstance it seems like in some cases it *won't* be the context for the current thread, so this would allocate the holder against the wrong thread in that case.

Reply:
Maybe `PreferedInstance` ? Please read the comments on top of `DefaultInstance` so that we could come up with better name.